gpytorch/kernels/kernel.py

```diff
@@ -94,7 +95,11 @@ def size(self, x1, x2):
 
     @abstractmethod
     def forward_diag(self, x1, x2, **params):
```

Comment:
The batch issues will happen for any lengthscale kernel (e.g. Periodic, Linear). We should probably remove the default `forward_diag` behavior.